File: lab2/lab2q2.py
import sys

line_counter = 0
sum_length = 0 
max_length = 0 
min_length = 0

for current_line in sys.stdin:

    stripped_line = current_line.strip()

    stripped_length = len( stripped_line )

    length_as_string = str( stripped_length )

    line_counter += 1

    sum_length = sum_length + stripped_length

    if stripped_length > max_length:
        max_length = stripped_length 
    elif min_length == 0:
        min_length = stripped_length 
    elif stripped_length < min_length:
        min_length = stripped_length 

    output_string = length_as_string + "\t" + current_line

    # Need to use an explicit end parameter otherwise it will output a \n
    print( output_string, end='' )

avg_length = sum_length / line_counter

avg_length_string = str( avg_length)
line_counter_string = str( line_counter)
sum_length_string = str ( sum_length)
max_length_string = str ( max_length)
min_length_string = str( min_length)

output_string2 = "lines:" + "\t" + line_counter_string + "\t" + " total length:" + "\t" + sum_length_string + "\t" + "average length:" +  "\t" + avg_length_string + "\t" + "max:" + "\t" +  max_length_string  + "\t" + "min: " +  "\t" + min_length_string

print( output_string2)

File: lab2/lab2q2.cpp
// clang++ -std=c++17 -Wall -Werror -Wextra -Wpedantic -g3 -o linelength linelength.cpp
#include <iostream>
#include <string>
#include <cctype>
#include <algorithm>

auto main() -> int
{
	std::string lineInput;

	auto line_counter {0};
	auto sum_length {0};
    auto max_length {0};
    auto min_length {0};

	while ( getline( std::cin, lineInput ) )
	{
		auto lineLength = lineInput.length();

		auto stringLineLength = std::to_string( lineLength );

        if (lineLength > max_length){
            max_length = lineLength; 

        } else if (min_length == 0) {
            min_length = lineLength;

        } else if (lineLength < min_length){
            min_length = lineLength ;
        }

		auto outputLine = stringLineLength + "\t" + lineInput;

  		std::cout << outputLine << std::endl; 

		line_counter = line_counter + 1;
		sum_length = sum_length + lineLength;
	}

auto avg_length = sum_length / static_cast<float>(line_counter);

auto outputLine2 =  std::string("lines:")  + std::string("\t" ) + std::to_string( line_counter )
+ std::string("\t") +  std::string("total length:") + std::string("\t") + std::to_string( sum_length ) 
+ std::string("\t") + std::string("averge length:") + std::string("\t") + std::to_string( avg_length )
+ std::string("\t") +std::string("max:") + std::string("\t") + std::to_string( max_length)
+ std::string("\t") +std::string("min:") + std::string("\t") + std::to_string( min_length);

std::cout << outputLine2 << std::endl;

}
